cypress/e2e/awx/administration/workflowApproval.cy.ts

```diff
@@ -0,0 +1,206 @@
+  let userWFApprove: User;
+  let userWFDeny: User;
+  let userWFCancel: User;
+  let inventory: Inventory;
+  let inventorySource: InventorySource;
+  let jobTemplate: JobTemplate;
+  let workflowJobTemplate: WorkflowJobTemplate;
+  let wfjobTemplateNode: WorkflowNode;
+  let approvalWFNode: WorkflowNode;
+
+  before(function () {
+    organization = this.globalOrganization as Organization;
+    project = this.globalProject as Project;
+    cy.awxLogin();
+
+    cy.createAwxUser(organization).then((u) => {
+      user = u;
+    });
+    cy.createAwxUser(organization).then((u) => {
+      userWFApprove = u;
+    });
+    cy.createAwxUser(organization).then((u) => {
+      userWFDeny = u;
+    });
+    cy.createAwxUser(organization).then((u) => {
+      userWFCancel = u;
+    });
+    cy.createAwxInventory({ organization: organization.id })
+      .then((i) => {
+        inventory = i;
+      })
+      .then(() => {
+        cy.createAwxInventorySource(inventory, project).then((invSrc) => {
+          inventorySource = invSrc;
+        });
+        cy.createAwxJobTemplate({
+          organization: organization.id,
+          project: project.id,
+          inventory: inventory.id,
+        }).then((jt) => (jobTemplate = jt));
+        cy.createAwxWorkflowJobTemplate({
+          organization: organization.id,
+          inventory: inventory.id,
+        }).then((wfjt) => {
+          workflowJobTemplate = wfjt;
+          cy.createAwxWorkflowVisualizerWJTNode(workflowJobTemplate).then((wfjtNode) => {
+            wfjobTemplateNode = wfjtNode;
+          });
+          cy.createAwxWorkflowVisualizerApprovalNode(workflowJobTemplate).then((appNode) => {
+            approvalWFNode = appNode;
+            cy.createWorkflowJTAlwaysNodeLink(approvalWFNode, wfjobTemplateNode);
+          });
+        });
+      });
+  });
+
+  beforeEach(() => {
+    cy.intercept(
+      'POST',
+      `api/v2/workflow_job_templates/${workflowJobTemplate.id.toString()}/launch`
+    ).as('launchWFJT');
+
+    cy.visit(`/templates/workflow_job_template/${workflowJobTemplate.id.toString()}/details`);
+    cy.getByDataCy('launch-template').click();
+  });
+
+  afterEach(() => {});
+
+  after(() => {
+    cy.deleteAwxWorkflowJobTemplate(workflowJobTemplate, { failOnStatusCode: false });
+    cy.deleteAwxJobTemplate(jobTemplate, { failOnStatusCode: false });
+    cy.deleteAwxInventorySource(inventorySource, { failOnStatusCode: false });
+    cy.deleteAwxInventory(inventory, { failOnStatusCode: false });
+    cy.deleteAwxUser(user, { failOnStatusCode: false });
+    cy.deleteAwxUser(userWFApprove, { failOnStatusCode: false });
+    cy.deleteAwxUser(userWFCancel, { failOnStatusCode: false });
+    cy.deleteAwxUser(userWFDeny, { failOnStatusCode: false });
+  });
+
+  function refreshToDelete(wfaID: string) {
+    // FIXME: the page needs to be refreshed to button be available
+    cy.visit(`/administration/workflow-approvals/`);
+    cy.filterTableByTextFilter('id', wfaID);
```

Comment:
Same comment as below? We could make this a cypress command?

Reply:
Might be, I want to be sure this will indeed need to be refreshed or if we can find a better way to wait for the workflow status. I also left it inside the test spec, since it only works for workflow approvals.